src/main/java/Duke.java

```diff
@@ -1,10 +1,185 @@
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.*;
+import javafx.scene.paint.Color;
+import javafx.stage.Stage;
+import javafx.scene.text.Font;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.shape.Circle;
+
+/**
+ * The main class that directs the required actions to the respective classes.
+ */
+public class Duke extends Application {
+    private Ui ui;
+    private Parser parser;
+    private Storage storage;
+    private TaskList tasks;
+
+    public Duke() {
+        this("data/duke.txt");
+    }
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            //ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/nobita.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/doraemon.png"));
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(user)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
+        dialogContainer.setSpacing(10);
+        //dialogContainer.setBackground(new Background(new BackgroundFill(Color.rgb(255,250,205),
+        //        CornerRadii.EMPTY, Insets.EMPTY)));
+        userInput.clear();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    //ignore hi and bye first, this takes in one input and churn a response
+    private String getResponse(String input) {
```

Comment:
This method seems a bit long, maybe you could separate out what to do in each case into different methods? For exmaple a handleList() method.